File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int mx = candies[0];
        for(int i = 0 ; i < candies.length; i++) {
            if(candies[i] > mx) {
                mx = candies[i];
            }
        }
        List<Boolean> res = new ArrayList();
        for(int i = 0 ; i < candies.length; i++) {
            int num = candies[i] + extraCandies;
            if(num >= mx) {
                res.add(true);
            } else res.add(false);
        }
        return res;
    }
}

File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> ans;
        int n = candies.size();
        int maxKids = *max_element(candies.begin(), candies.end());

        for(int i= 0; i < n; i++) {
            int mxCandies = candies[i] + extraCandies;
            if (mxCandies >= maxKids) {
                ans.push_back(true);
            } else {
                ans.push_back(false);
            }
        }
        
        return ans;
    }
};